src/protobuf_definitions/meadowgrid/meadowgrid.proto

```diff
@@ -236,48 +255,47 @@ message ProcessState {
 
         // The meadowgrid coordinator has received the Job
         RUN_REQUESTED = 1;
-        // The meadowgrid coordinator has assigned the Job to a worker
-        // TODO add worker_id
-        ASSIGNED = 2;
```

Comment:
Right now, we have three services involved: meadowflow -> meadowgrid coordinator -> meadowgrid agent. Currently, the workflow is:
- meadowflow creates a request id and stores it (in memory), but does NOT decide which meadowgrid agent will run the job)
- meadowflow sends the request to the meadowgrid coordinator (it's safe to send duplicate requests with the same id, subsequent requests to the coordinator with the same id will be ignored: https://github.com/meadowdata/meadowdata/blob/main/src/protobuf_definitions/meadowgrid/meadowgrid.proto#L156)
- the coordinator decides which agent will run the job and stores that (in memory)
- the agent runs the job
So I think the scenario you're talking about works okay (other than the fact that all the storing is only in memory right now)?

This change doesn't "get rid of" the coordinator keeping track of which jobs are assigned to which agents, it just makes it more complicated, because now there's both "pending" (coordinator has decided on an agent for the job, but the agent doesn't know it yet) and "assigned" (the agent has "picked up" the pending job, but it hasn't started running yet).